### PR TITLE
Add dummy logger for internally disabling logging for some features

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/1828 and https://github.com/PyTorchLightning/pytorch-lightning/issues/1830
Make sure that users code can run during tuner algorithms (learning rate finder and batch scaler) when they call loggers directly in their code.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by williamFalcon** [comment]
awesome!
in the case where i did this:

```
self.logger.experiment.add_image('images', grid, 0)
```

would the no-op still let any arbitrary function work like add_image since it's not defined? we basically need a way to allow any call here no?

**Comment by williamFalcon** [comment]
Never mind lol... answered myself haha.

Well done!

image
